src/openrct2/ride/RideConstruction.cpp

```diff
@@ -654,7 +659,8 @@ void ride_construction_set_default_next_piece()
                 }
             }
 
-            curve = gTrackCurveChain[trackType].next;
+            ted = &GetTrackElementDescriptor(trackType);
```

Comment:
Any reason you opted for a pointer instead of a reference?

Reply:
The variable gets reassigned.